### ISSUE TITLE
[skia] build failure

### ISSUE THREAD

**Issue by PatrykMiszczak** [microsoft/vcpkg#23135]
**Host Environment**
- OS: MacOS (x86_64-apple-darwin20.6.0)
- Compiler: Apple clang version 12.0.5

**To Reproduce**
```
cmake -B build -S .
cmake --build build
```

part of `CMakeList.txt` related to `skia`
```
find_package(skia CONFIG REQUIRED)
target_link_libraries(dsptool PRIVATE skia skia::skia)
```

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.

The first one (I've fixed this by myself):
```
CMake Error at build/vcpkg_installed/x64-osx/share/skia/skiaConfig.cmake:1:
  Parse error.  Expected a newline, got identifier with text "add_library".
Call Stack (most recent call first):
  /Users/patryk/DSP/playground/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package)
  CMakeLists.txt:23 (find_package)


CMake Error at /Users/patryk/DSP/playground/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  _find_package Error reading CMake code from
  "/Users/patryk/DSP/playground/toolkit/build/vcpkg_installed/x64-osx/share/skia/skiaConfig.cmake".
Call Stack (most recent call first):
  CMakeLists.txt:23 (find_package)
```

The second one:
https://gist.github.com/PatrykMiszczak/30f06eaa70c535711c6d51bbd0a21774

-Please attach any additional failure logs mentioned in the console output.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

The first error I fixed by manually fixing `skiaConfig.cmake` file. For some reason, the first line of this file is invalid:
```
get_filename_component(VCPKG_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)add_library(skia INTERFACE)
```

The error can be resolved with adding new line before `add_library`:
```
get_filename_component(VCPKG_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)
add_library(skia INTERFACE)
```

But I still have a problem with the linker errors. It looks like `target_link_libraries(dsptool PRIVATE skia skia::skia)` doesn't pass all dependencies to the linker. I've checked what libraries are linked using verbose mode in cmake (`cmake --build build -v`) and I found in logs this command:
```
/usr/local/opt/llvm/bin/clang++  -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/dsptool.dir/src/main.cpp.o CMakeFiles/dsptool.dir/src/commands/generate.cpp.o CMakeFiles/dsptool.dir/src/commands/impulseresponse.cpp.o CMakeFiles/dsptool.dir/src/commands/listparams.cpp.o CMakeFiles/dsptool.dir/src/commands/play.cpp.o CMakeFiles/dsptool.dir/src/commands/process.cpp.o CMakeFiles/dsptool.dir/src/helpers/analyzer.cpp.o CMakeFiles/dsptool.dir/src/helpers/kernelloader.cpp.o CMakeFiles/dsptool.dir/src/helpers/kernelparamsparser.cpp.o CMakeFiles/dsptool.dir/src/helpers/kernelprocessor.cpp.o CMakeFiles/dsptool.dir/src/helpers/player.cpp.o CMakeFiles/dsptool.dir/src/helpers/visualizer.cpp.o -o dsptool  vcpkg_installed/x64-osx/debug/lib/libfmtd.a vcpkg_installed/x64-osx/debug/lib/libsfml-audio-s-d.a -framework OpenAL vcpkg_installed/x64-osx/debug/lib/libsfml-system-s-d.a -stdlib=libc++ -lpthread vcpkg_installed/x64-osx/lib/libvorbisfile.a vcpkg_installed/x64-osx/lib/libvorbisenc.a vcpkg_installed/x64-osx/lib/libvorbis.a vcpkg_installed/x64-osx/lib/libogg.a vcpkg_installed/x64-osx/lib/libFLAC.a vcpkg_installed/x64-osx/lib/libskia.a
```

I tried to run this command manually (from `build` directory) with some modifications and I found that after adding:
```
-framework Foundation -framework AppKit vcpkg_installed/x64-osx/lib/libpng.a vcpkg_installed/x64-osx/lib/libwebp.a vcpkg_installed/x64-osx/lib/libwebpdecoder.a vcpkg_installed/x64-osx/lib/libwebpdemux.a vcpkg_installed/x64-osx/lib/libwebpmux.a vcpkg_installed/x64-osx/lib/libjpeg.a vcpkg_installed/x64-osx/lib/libz.a
```
all the symbols are resolved correctly, so I guess, there is some problem with generating the list of dependencies.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. I can't reproduce this issue locally. I use sample project to reproduce this issue, but I build succeed. This is my project information.
**CMakeLists.txt**
```
cmake_minimum_required (VERSION 3.9)

project (test001 C CXX)

add_executable(test001 main.cpp)

find_package(skia CONFIG REQUIRED)
target_link_libraries(test001 PRIVATE skia skia::skia)
```
**vcpkg.json**
```
{
  "name": "test001",
  "version-string": "0.0.1",
  "dependencies": [
    "skia"
  ]
}
```
**main.cpp**
```
#include <iostream>

int main()
{
    std::cout << "Hello World!\n";
}
```
**Build process:**
`../vcpkg/downloads/tools/cmake-3.21.1-osx/cmake-3.21.1-macos-universal/CMake.app/Contents/bin/cmake -DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/lily/vcpkg/scripts/buildsystems/vcpkg.cmake CMakeLists.txt`
![image](https://user-images.githubusercontent.com/94091114/155093556-f4d68835-b78b-4270-85b6-2e77e48100ff.png)

![image](https://user-images.githubusercontent.com/94091114/155093395-817a4106-dee4-4058-ba4e-3cbdd923ed09.png)
![image](https://user-images.githubusercontent.com/94091114/155093430-bb994014-a07f-4fed-a5d9-0efc9e0c3b51.png)

**Comment by LilyWangLL** [comment]
Ping @PatrykMiszczak for response.

**Comment by dg0yt** [comment]
This was fixed by #22236. You need to install the updated `vcpkg-cmake-config` port for the host triplet.
(And you will need to rebuild all depending ports.)

**Comment by PatrykMiszczak** [comment]
Sorry for the late response. 
@dg0yt  - as it comes to the first error, you're probably right. There was a problem with:
```
get_filename_component(VCPKG_IMPORT_PREFIX
```

**Comment by PatrykMiszczak** [comment]
Sorry for the late response.

@dg0yt  - as it comes to the first error, you're probably right. I see the PR you've sent corresponds to exactly the same problem.

@LilyWangLL - I'll prepare some example source code. I think the build doesn't fail in your case because there are no references to the library. Please note that the problem occurs at the linking stage

**Comment by PatrykMiszczak** [comment]
@LilyWangLL - can you try with this source code?

```c++
#include "skia/core/SkData.h"
#include "skia/core/SkSurface.h"
#include "skia/core/SkCanvas.h"
#include "skia/core/SkPath.h"
#include "skia/core/SkImage.h"
#include "skia/core/SkStream.h"
#include "skia/core/SkPaint.h"

void draw(SkCanvas* canvas);

int main() {
	sk_sp<SkSurface> surface;
	SkCanvas* canvas;

	surface = SkSurface::MakeRasterN32Premul(100, 100);
	canvas = surface->getCanvas();

	draw(canvas);

	sk_sp<SkImage> img(surface->makeImageSnapshot());
	sk_sp<SkData> png(img->encodeToData());

	SkFILEWStream out("output.png");
	out.write(png->data(), png->size());
}

void draw(SkCanvas* canvas) {
	canvas->save();
	canvas->translate(SkIntToScalar(128), SkIntToScalar(128));
 	canvas->rotate(SkIntToScalar(45));
	SkRect rect = SkRect::MakeXYWH(-90.5f, -90.5f, 181.0f, 181.0f);
	SkPaint paint;
	paint.setColor(SK_ColorBLUE);
	canvas->drawRect(rect, paint);
	canvas->restore();
}
```

**Comment by LilyWangLL** [comment]
Thanks for @PatrykMiszczak's reply, I will double confirm soon.

**Comment by MonicaLiu0311** [comment]
Currently, https://github.com/microsoft/vcpkg/pull/22236 has been merged and skia has been upgraded to [0.36.0#7](https://github.com/microsoft/vcpkg/blob/master/ports/skia/vcpkg.json#L3-L4); so I will close this issue, if it is still an issue, please reopen it.
